### PR TITLE
Do not import mail already seen

### PR DESCRIPTION
**Pull request by stonebuzz**
Prevent loop when message is read but when it's impossible to move or delete it
due to mailbox problem (ie: full)

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #number

### CODE REVIEW

**Comment by cedric-anne** [comment]
Even if it may not be a common case, it may happen that users expect read emails to be collected too.

An alternative solution would be to create a `CommonDBChild` (e.g. `glpi_ticketsource`) where we would store:
 - `mailcollectors_id`: ID of the mailcollector that retrieve the message,
 - `tickets_id`: ID of the ticket that was created for this message,
 - `messageid`: value of `Message-ID` headerof the email.

With this, we would be able to check if a ticket was already created for a given `Message-ID`, and skip it in this case.

Thus, this new object could be used in the future to store the raw source of the message, if we found how to fetch it correctly, in order to help debugging, add a "source email view/export" feature, ...

**Comment by orthagh** [comment]
just `messageid` in glpi_tickets should be sufficient, no ?
maybe also `mailcollectors_id` if needed, but the relation between a mail and a ticket is 1 <-> 1, no ?

**Comment by orthagh** [comment]
for FLAG_SEEN, the easy solution could be simply implements an option in the mailcollector, and let the administrator decides what to do with their mailgate

**Review by trasher — APPROVED** [review]

File: tests/imap/MailCollector.php [hunk 1]
```diff
@@ -198,17 +199,18 @@ private function doConnect() {
...
+         'login'                 => 'testuser',
+         'is_active'             => true,
+         'passwd'                => 'applesauce',
+         'mail_server'           => '127.0.0.1',
+         'server_type'           => '/imap',
+         'server_port'           => 143,
+         'server_ssl'            => '',
+         'server_cert'           => '/novalidate-cert',
+         'add_cc_to_observer'    => 1,
+         'collect_only_unread'   => 1, //add ccuser as observer in ticket
```
Comment (cedric-anne) [inline tests/imap/MailCollector.php:212]:
```suggestion
         'collect_only_unread'   => 1,
```

**Review by cedric-anne — COMMENTED** [review]
OK for me.

Should we target 9.5 or master branch ?

**Comment by cedric-anne** [comment]
> An alternative solution would be to create a `CommonDBChild` (e.g. `glpi_ticketsource`) where we would store:
> 
> * `mailcollectors_id`: ID of the mailcollector that retrieve the message,
> * `tickets_id`: ID of the ticket that was created for this message,
> * `messageid`: value of `Message-ID` headerof the email.
> 
> With this, we would be able to check if a ticket was already created for a given `Message-ID`, and skip it in this case.
> 
> Thus, this new object could be used in the future to store the raw source of the message, if we found how to fetch it correctly, in order to help debugging, add a "source email view/export" feature, ...

As discussed on internal meeting, this will not be done for now but could be done in another pull request.

**Comment by trasher** [comment]
Targetting 9.5 should be ok, as far as I remember, this is to fix a behavior that has changed when switching from php-imap to laminas mail.
@stonebuzz could you please rebase to fix conflicts? Thanks!

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**